app/js/shortID.js

```diff
@@ -0,0 +1,4 @@
+module.exports = function generate(str) {
+	return (
+		str.concat(String(Math.floor(Math.random() * ((9999999 - 9) + 9)))));
```

Comment:
yeah creating UUIDs in JS is tough. What you have here will probably work just fine.
There are npm packages for node that might work in browser - check out https://www.npmjs.com/package/uuid although the Issues queue talks about issues with babel - shrug.